.changeset/forty-kids-cut.md

```diff
@@ -0,0 +1,19 @@
+---
+'@kaizen/components': minor
+---
+
+Wrap component styles in a CSS Cascade Layer
+
+To unblock adoption of Tailwind CSS v4 in consuming apps, Kaizen component styles must be wrapped in `@layer components` that can be given lower priority than Tailwind's new `@layer utilities`.
+
+Ideally this would not be necessary, and consuming apps could simply import Kaizen's styles into a layer:
+
+```
+@import url(~@kaizen/components/dist/styles.css) layer(components);
+```
+
+Unfortunately, this syntax is [not supported](https://github.com/vercel/next.js/issues/55763#issuecomment-2791540126) by the outdated version of webpack css-loader that is contained in Next.js. Until it is, we must publish styles for packages like Kaizen with a layer provided.
+
+#### Isn't this a breaking change?
+
+Although technically this lowers the priority of Kaizen's component styles below any consuming apps' "unlayered" styles, regardless of selector specificity or source order, we believe that Kaizen styles are already output before app styles in apps' CSS bundles to ensure app styles take priority; therefore, this should have no effect on the relative priority of styles in consuming apps.
```

Comment:
Is there any chance that vendor styles might start winning out the specificity wars / is something we should flag when this gets released? 

I recall we had an [issue in mumur](https://cultureamp.slack.com/archives/C0189KBPM4Y/p1737687211324029) a while back with bootstrap supersede some of our Link styles but not sure if this would be an issue in the new world of CSS layers 🍰

Reply:
Hmm yes, it's definitely a risk. Un-layered Bootstrap styles in Murmur will have greater priority than any styles in Kaizen following this change. Are you aware of any specific styles in Kaizen that are there to override Bootstrap in Murmur? If so, we could consider pulling them out of the `components` layer.

Reply:
To my knowledge we don't have anything specific (least not from that). I think in that instance it was recommended that the team remove the styles that impact link

Reply:
If such styles needed to stay in the consuming app, the other fix would be for the consuming app to move them into the `components` layer too.